delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/Archives.scala

```diff
@@ -28,19 +199,19 @@ object Archives {
   ): IO[ArchiveRejection, ArchiveCreated] = {
 
     def checkPathDuplication: IO[ArchiveRejection, Unit] = {
-      val duplicates = command.resources.value
+      val duplicates = command.value.resources.value
         .groupBy(_.path)
         .collect {
-          case (path, refs) if refs.size > 1 => path
+          case (Some(path), refs) if refs.size > 1 => path
         }
         .toSet
       if (duplicates.nonEmpty) IO.raiseError(DuplicateResourcePath(duplicates))
       else IO.unit
     }
 
     def checkAbsolutePaths: IO[ArchiveRejection, Unit] = {
-      val nonAbsolute = command.resources.value.foldLeft(Set.empty[Path]) { case (set, elem) =>
-        if (!elem.path.isAbsolute) set + elem.path else set
+      val nonAbsolute = command.value.resources.value.map(_.path).collect {
+        case Some(value) if !value.isAbsolute => value
       }
       if (nonAbsolute.nonEmpty) IO.raiseError(PathIsNotAbsolute(nonAbsolute))
```

Comment:
This is not necessary since it is already present in JsonLdDecoder

Reply:
I prefer to have the check also here if it's already implemented and tested; if the archive will be used in a different way in the future it will not crash unexpectedly.

Reply:
The right way to do it would be to provide a proper constructor for `ResourceReference` and `FileReference` and keep to default ones private. Those constructors should check for the passed `Path` to be absolute and fail otherwise.

It makes sense to fail early during `ArchiveReference` construction instead of to have to fail on the state machine.

Reply:
That's probably better and achieves the same goal...

Reply:
Actually the best way would be to have this in the sdk:
```scala
final case class AbsolutePath private (value: Path)
object AbsolutePath {
  def apply(path: Path): Either[FormatError, AbsolutePath] = ???
  implicit val absolutePathEncoder: Encoder[AbsolutePath] = ???
  implicit val absolutePathJsonLdEncoder: JsonLdEncoder[AbsolutePath] = ???
}
```

Since then you can use it in your `ResourceReference` and I can use it in `DiskStorageFields`.